pkg/gui/controllers/tags_controller.go

```diff
@@ -131,6 +131,9 @@ func (self *TagsController) remoteDelete(tag *models.Tag) error {
 				Prompt: confirmPrompt,
 				HandleConfirm: func() error {
 					return self.c.WithWaitingStatus(self.c.Tr.DeletingStatus, func(t gocui.Task) error {
+						self.c.State().SetRefOperation(tag.FullRefName(), types.RefOperationDeleting, context.TAGS_CONTEXT_KEY)
```

Comment:
We always defer a clear right after setting the ref and we always pass the same arguments to it. We can DRY this up: I'm thinking we have a function like WithWaitingStatus which takes a function as an argument and does the set/clear around calling that function.

Reply:
Good idea, see a0510f5b285.